src/Workspaces/Core/Portable/SymbolKey/SymbolKey.ErrorTypeSymbolKey.cs

```diff
@@ -13,9 +15,23 @@ private static class ErrorTypeSymbolKey
             public static void Create(INamedTypeSymbol symbol, SymbolKeyWriter visitor)
             {
                 visitor.WriteString(symbol.Name);
-                visitor.WriteSymbolKey(symbol.ContainingSymbol as INamespaceOrTypeSymbol);
-                visitor.WriteInteger(symbol.Arity);
+                switch (symbol.ContainingSymbol)
+                {
+                    case INamedTypeSymbol parentType:
+                        visitor.WriteInteger(0);
+                        visitor.WriteSymbolKey(parentType);
+                        break;
+                    case INamespaceSymbol parentNamespace:
+                        visitor.WriteInteger(1);
+                        visitor.WriteStringArray(GetContainingNamespaceNames(parentNamespace));
+                        break;
+                    default:
+                        visitor.WriteInteger(2);
+                        visitor.WriteSymbolKey(null);
```

Comment:
note: i technically don't need this value (since the reader can just skip reading it in the '2' case). however, it makes the signatures a lot easier to read if they're consistent with the number of children they write out.

Reply:
Can you add this comment to the file? Your explanation is good but it'd be more helpful to the future if it's in the code.

Reply:
sure.